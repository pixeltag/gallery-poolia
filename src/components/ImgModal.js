import closeIcon from "../assets/svg/close.svg";

const ImgModal = ({
  full,
  description,
  title,
  modalStatus,
  handleCloseModalCb,
}) => {
  const handleCloseModal = () => {
    handleCloseModalCb();
  };
  return (
    <div data-testid="modal" className={modalStatus ? "modal show" : "modal"}>
      <button onClick={() => handleCloseModal()} className="close-btn">
        <img src={closeIcon} alt="Close modal" />
      </button>
      <img data-testid="img" className="modal-img" src={full} alt={title} />
      <div className="modal-body">
        <h3 data-testid="title">{title ? title : "Image title"}</h3>
        <p data-testid="description">{description}</p>
      </div>
    </div>
  );
};

export default ImgModal;
