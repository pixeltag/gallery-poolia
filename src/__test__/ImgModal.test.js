import { render, cleanup } from "@testing-library/react";
import { unmountComponentAtNode } from "react-dom";

import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import ImgModal from "../components/ImgModal";

configure({ adapter: new Adapter() });

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  cleanup();
  container.remove();
  container = null;
});

const fakeImage = {
  title: "image titile",
  src: "url-src",
  full: "url-full-src",
  description: "image description",
};

test("<ImgModal /> component renders correctly", async () => {
  const container = render(<ImgModal />);
  const modalElement = await container.findByTestId("modal");
  expect(modalElement).toBeInTheDocument();
});

test("<ImgModal /> component renders image data", async () => {
  const container = render(<ImgModal {...fakeImage} />);
  const imgElement = await container.findByTestId("img");
  const titleElement = await container.findByTestId("title");
  const descriptionElement = await container.findByTestId("description");
  expect(imgElement.getAttribute("src")).toBe(fakeImage.full);
  expect(imgElement.getAttribute("alt")).toBe(fakeImage.title);
  expect(titleElement.textContent).toBe(fakeImage.title);
  expect(descriptionElement.textContent).toBe(fakeImage.description);
});

test("<ImgModal /> appears when modalStatus true", async () => {
  const container = render(<ImgModal {...fakeImage} modalStatus={true} />);
  const modalElement = await container.findByTestId("modal");
  expect(modalElement.getAttribute("class")).toBe("modal show");
});

test("<ImgModal /> disappears when modalStatus false", async () => {
  const container = render(<ImgModal {...fakeImage} modalStatus={false} />);
  const modalElement = await container.findByTestId("modal");
  expect(modalElement.getAttribute("class")).toBe("modal");
});

test("<ImgModal /> click the close btn fire function callback", async () => {
  const closeModal = jest.fn();
  const container = shallow(<ImgModal handleCloseModalCb={closeModal} />);
  const closeBtn = await container.find('button');
  closeBtn.simulate("click");
  expect(closeModal.mock.calls.length).toEqual(1);
});

test("ImgModal component should match the snapshot", () => {
  const wrapper = shallow(<ImgModal />);
  expect(wrapper.html()).toMatchSnapshot();
  cleanup();
});
