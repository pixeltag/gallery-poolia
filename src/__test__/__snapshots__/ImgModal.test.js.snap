// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImgModal component should match the snapshot 1`] = `"<div data-testid=\\"modal\\" class=\\"modal\\"><button class=\\"close-btn\\"><img src=\\"close.svg\\" alt=\\"Close modal\\"/></button><img data-testid=\\"img\\" class=\\"modal-img\\"/><div class=\\"modal-body\\"><h3 data-testid=\\"title\\">Image title</h3><p data-testid=\\"description\\"></p></div></div>"`;
